network-operator may crash if sriov-device-plugin deployed

network-operator may crash if sriov-device-plugin deployed

Network operator will crash if sriov-device-plugin deployment
requested but there are no nodes with `feature.node.kubernetes.io/pci-15b3.present`
label in the cluster.


**What happened**:
panic in network-operator

**What you expected to happen**:

no panic

**How to reproduce it (as minimally and precisely as possible)**:

Deploy network-operator with sriov-device-plugin on cluster which has not mellanox nics

Log:

```
2021-10-28T09:13:59.576Z        INFO    state   Sync Custom resource    {"State:": "state-SRIOV-device-plugin", "Name:": "nic-cluster-policy", "Namespace:": ""}
E1028 09:13:59.576598       1 runtime.go:78] Observed a panic: runtime.boundsError{x:0, y:0, signed:true, code:0x0} (runtime error: index out of range [0] with length 0)                                      
goroutine 224 [running]:                                                                                                                                                                                       
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1694be0, 0xc0000c9f50)                                                                                                                                         
        /go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:74 +0x95                                                                                                                           
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)                                                                                                                                                
        /go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:48 +0x86                                                                                                                           
panic(0x1694be0, 0xc0000c9f50)                                                                                                                                                                                 
        /usr/local/go/src/runtime/panic.go:965 +0x1b9                                                                                                                                                          
github.com/Mellanox/network-operator/pkg/state.(*stateSriovDp).getManifestObjects(0xc0004c9770, 0xc0006fa1a0, 0x19415e0, 0xc0004cc2d0, 0x6, 0x6, 0xc000261880, 0x148fb1c, 0xc000138630)
        /workspace/pkg/state/state_sriov_dp.go:134 +0x5d3                                                                                                                                                      
github.com/Mellanox/network-operator/pkg/state.(*stateSriovDp).Sync(0xc0004c9770, 0x173ba00, 0xc0006fa1a0, 0x1948818, 0xc000402148, 0x4, 0x1770a2d, 0x5, 0x1779c06)
        /workspace/pkg/state/state_sriov_dp.go:90 +0x265                                                                                                                                                       
github.com/Mellanox/network-operator/pkg/state.(*Group).Sync(0xc000261a50, 0x173ba00, 0xc0006fa1a0, 0x1948818, 0xc000402148, 0x2, 0x4, 0x8)                                                                    
        /workspace/pkg/state/group.go:43 +0x2c7                                                                                                                                                                
github.com/Mellanox/network-operator/pkg/state.(*stateManager).SyncState(0xc0003c6ba0, 0x173ba00, 0xc0006fa1a0, 0x1948818, 0xc000402148, 0x0, 0x0, 0x0, 0x0, 0x0, ...)                                         
        /workspace/pkg/state/manager.go:92 +0x28a                                                                                                                                                              
github.com/Mellanox/network-operator/controllers.(*NicClusterPolicyReconciler).Reconcile(0xc00048d480, 0x19722f8, 0xc000482360, 0x0, 0x0, 0xc0000bd200, 0x12, 0xc000482360, 0xc000030000, 0x164e280, ...)      
        /workspace/controllers/nicclusterpolicy_controller.go:119 +0x730                                                                                                                                       
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0004b57c0, 0x1972250, 0xc0006c8ac0, 0x16186c0, 0xc00046e320)                                                          
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.1/pkg/internal/controller/controller.go:297 +0x30d                                     
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0004b57c0, 0x1972250, 0xc0006c8ac0, 0xc00060f600)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.1/pkg/internal/controller/controller.go:252 +0x205                                                                                                     
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2(0x1972250, 0xc0006c8ac0)                                                                                                    
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.1/pkg/internal/controller/controller.go:215 +0x4a                                                                                                      
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()                                                                                                                                               
        /go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185 +0x37                                                                                                                                
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc00060f750)                                                                                                                                             
        /go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:155 +0x5f                                                                                                                                
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000261f50, 0x1942e40, 0xc0004822d0, 0xc0006c8a01, 0xc0000467e0)                              
        /go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:156 +0x9b                                                                                                                                
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00060f750, 0x3b9aca00, 0x0, 0x9a8701, 0xc0000467e0)                                                                                                           
        /go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:133 +0x98                                                                                                                                
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x1972250, 0xc0006c8ac0, 0xc00037c150, 0x3b9aca00, 0x0, 0x1)                         
        /go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185 +0xa6                                                                                                                                
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(0x1972250, 0xc0006c8ac0, 0xc00037c150, 0x3b9aca00)                                                                                                          
        /go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:99 +0x57                                                                                                                                 
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1                                                                                                                    
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.1/pkg/internal/controller/controller.go:212 +0x40d                                      
panic: runtime error: index out of range [0] with length 0 [recovered]                                                                                                                                         
        panic: runtime error: index out of range [0] with length 0       
```

--- a/pkg/state/state_sriov_dp.go
+++ b/pkg/state/state_sriov_dp.go
@@ -125,6 +125,11 @@ func (s *stateSriovDp) getManifestObjects(
 	nodeInfo nodeinfo.Provider) ([]*unstructured.Unstructured, error) {
 	attrs := nodeInfo.GetNodesAttributes(
 		nodeinfo.NewNodeLabelFilterBuilder().WithLabel(nodeinfo.NodeLabelMlnxNIC, "true").Build())
+	if len(attrs) == 0 {
+		log.V(consts.LogLevelInfo).Info("No nodes with NVIDIA NICs where found in the cluster.")
+		return []*unstructured.Unstructured{}, nil
+	}
+
 	renderData := &sriovDpManifestRenderData{
 		CrSpec:              cr.Spec.SriovDevicePlugin,
 		NodeAffinity:        cr.Spec.NodeAffinity,

